client/dashboard/me/billing-purchases/purchase-settings/index.tsx

```diff
@@ -0,0 +1,1209 @@
+import {
+	__experimentalText as Text,
+	__experimentalVStack as VStack,
+	__experimentalHStack as HStack,
+	__experimentalHeading as Heading,
+	DropdownMenu,
+	MenuGroup,
+	MenuItem,
+	Card,
+	CardBody,
+	Button,
+	Icon,
+	ToggleControl,
+	Notice,
+	ExternalLink,
+} from '@wordpress/components';
+import { DataForm } from '@wordpress/dataviews';
+import { createInterpolateElement } from '@wordpress/element';
+import { __, _n, isRTL, sprintf } from '@wordpress/i18n';
+import {
+	moreVertical,
+	chevronLeft,
+	chevronRight,
+	calendar,
+	currencyDollar,
+	siteLogo,
+	commentAuthorAvatar,
+} from '@wordpress/icons';
+import { useAnalytics } from '../../../app/analytics';
+import { useAuth } from '../../../app/auth';
+import { useLocale } from '../../../app/locale';
+import { domainsQuery } from '../../../app/queries/domains';
+import { purchaseQuery, userPurchaseSetAutoRenewQuery } from '../../../app/queries/me-purchases';
+import { siteBySlugQuery } from '../../../app/queries/site';
+import { siteDifmWebsiteContentQuery } from '../../../app/queries/site-do-it-for-me';
+import { siteJetpackKeysQuery } from '../../../app/queries/site-jetpack-keys';
+import { reinstallMarketplacePluginsQuery } from '../../../app/queries/site-marketplace';
+import { purchaseSettingsRoute } from '../../../app/router/me';
+import { ActionList } from '../../../components/action-list';
+import ClipboardInputControl from '../../../components/clipboard-input-control';
+import { useFormattedTime } from '../../../components/formatted-time';
+import { PageHeader } from '../../../components/page-header';
+import PageLayout from '../../../components/page-layout';
+import {
+	ProductUpgradeMap,
+	AkismetUpgradesProductMap,
+	SubscriptionBillPeriod,
+	DomainProductSlugs,
+	useMyDomainInputMode,
+	WPCOM_DIFM_LITE,
+	OFFSITE_REDIRECT,
+} from '../../../data/constants';
+import { DomainTransferStatus } from '../../../data/domains';
+import { formatDate } from '../../../utils/datetime';
+import { getEmailManagementPath } from '../../../utils/email-paths';
+import {
+	getBillPeriodLabel,
+	getTitleForDisplay,
+	getSubtitleForDisplay,
+	isExpiring,
+	isExpired,
+	isRenewing,
+	isIncludedWithPlan,
+	isOneTimePurchase,
+	isMarketplaceTemporarySitePurchase,
+	isMarketplacePlugin,
+	isJetpackTemporarySitePurchase,
+	isAkismetProduct,
+	isJetpackCrmProduct,
+	isTitanMail,
+	isGoogleWorkspace,
+} from '../../../utils/purchase';
+import { encodeProductForUrl } from '../../../utils/wpcom-checkout';
+import { PurchasePaymentMethod } from '../purchase-payment-method';
+import { getPurchaseUrlForId, getAddPaymentMethodUrlFor } from '../urls';
+import type { User } from '../../../data/me';
+import type { Purchase } from '../../../data/purchase';
+import type { Field } from '@wordpress/dataviews';
+import type { ReactNode, ReactElement } from 'react';
+
+import './style.scss';
+
+function getServicePathForCheckoutFromPurchase( purchase: Purchase ): string {
```

Comment:
Might worth extracting this logic because it's already used here

https://github.com/Automattic/wp-calypso/blob/eee2e436affe24a997a61af2bc0ca25f8abeacdc/client/dashboard/utils/domain.ts#L25-L30

Reply:
Good find! I've deduplicated that now.